### PR TITLE
fix times method when also using the with method

### LINKED ISSUE #44: ->times() breaks most functionality added to factory
```
UserFactory::new()
    ->with(\App\Review::class, 'reviews', 10)
    ->times(10)
    ->create();
```

**Expected**
10 users with 10 reviews each

**Actual**
10 users with 0 reviews

It looks like `->times()` is just getting the attribute data and creating models from that, but the factories might have additional methods, properties, and logic in the create/make methods. Might be able to fix this and simplify the code by cloning the factory X times, passing it to `CollectionFactory`, and then calling create/make on each clone individually. Any thoughts or concerns before I work on a PR in the next few days?

### PR DESCRIPTION
**Pull request by JoshMoreno**
Fixes #44 
You took all the good names so I wasn't sure what to call the new class that handles multiple factories! haha. 

All tests are passing and everything seems to be behaving the same.

### CODE REVIEW

**Comment by christophrumpel** [comment]
Looks good so far, it also seems we don't need my "CollectionFactory" anymore. Then you could also use the name.

File: src/MultiFactoryCollection.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+class MultiFactoryCollection{
+    protected Collection $factories;
+
+    public function __construct(Collection $factories)
+    {
+        $this->factories = $factories;
+    }
+
+    public function create()
```
Comment (christophrumpel) [inline src/MultiFactoryCollection.php]: Can you please add return types to "create" and "make" method.

File: src/BaseFactory.php [hunk 2]
```diff
@@ -40,37 +40,35 @@ protected function build(array $extra = [], string $creationType = 'create')
         $modelData = $this->prepareModelData($creationType, $this->getDefaults($this->faker));
         $model = $this->modelClass::$creationType(array_merge($modelData, $this->overwriteDefaults, $extra));
 
-        if ($this->relatedModels->isEmpty()) {
+        if ($this->relatedModelFactories->isEmpty()) {
             return $model;
         }
 
+        $relatedModels = $this->relatedModelFactories->map(fn($factory) => $factory->make());
```
Comment (christophrumpel) [inline src/BaseFactory.php]: here we could also use the high order version, right? `->map->make();`

File: src/MultiFactoryCollection.php (hunk 1 above)
Comment (JoshMoreno) [reply src/MultiFactoryCollection.php]: Whoops. Done.

File: src/BaseFactory.php (hunk 2 above)
Comment (JoshMoreno) [reply src/BaseFactory.php]: Yup, I always forget about those. Refactored BaseFactory and MultiFactoryCollection to use these

**Comment by JoshMoreno** [comment]
I didn't change the name since I thought `CollectionFactory` sounded like it makes collections, but we expect a collection to be passed in via the constructor. I can change if you prefer `CollectionFactory` though.

**Comment by christophrumpel** [comment]
Thanks, ok makes sense too. Let's leave it like that. Thanks a lot for the PR and the fix 👍

**Pull request merged**